eng/pipelines/templates/jobs/archetype-sdk-integration.yml

```diff
@@ -92,18 +122,40 @@ jobs:
       - script: |
           node common/scripts/install-run-rush.js install
         displayName: "Install dependencies"
+        condition: eq(variables['DependencyVersion'],'repo')
 
       # Option "-p max" ensures parallelism is set to the number of cores on all platforms, which improves build times.
       # The default on Windows is "cores - 1" (microsoft/rushstack#436).
       - script: |
           node common/scripts/install-run-rush.js build -t "${{parameters.PackageName}}" --verbose -p max
         displayName: "Build libraries"
+        condition: eq(variables['DependencyVersion'],'repo')
 
       # Option "-p max" ensures parallelism is set to the number of cores on all platforms, which improves build times.
       # The default on Windows is "cores - 1" (microsoft/rushstack#436).
       - script: |
           node common/scripts/install-run-rush.js build:test -t "${{parameters.PackageName}}" --verbose -p max
         displayName: "Build test assets"
+        condition: eq(variables['DependencyVersion'],'repo')
+
+      - script: |
+          npm install
+          node .\index.js --artifact-name "${{parameters.PackageName}}" --version-type $(DepLevel)
+        displayName: "Install dependencies"
+        workingDirectory: $(Build.SourcesDirectory)/eng/tools/dependencyTesting
+        condition: ne(variables['DependencyVersion'],'repo')
+
+      - script: |
```

Comment:
To help simply I'd combine these steps into one script step. So we can pull the setup and building of the tests into one step.